[PowerPC] Update DMF VSX ACC data transfer instructions

lei137 opened the pull request:
For cpu=future, acc registers no longer overlap VSRs and are prefixed with `dm`.  The original, xxmfacc/xxmtacc instructions are now extended menemonics to it's dm* equivalents.

lei137 committed:
[PowerPC] Update DMF VSX ACC data transfer instructions

For cpu=future, acc registers no longer overlap VSRs and
are prefixed with `dm`.  The original, xxmfacc/xxmtacc instructions
are now extended menemonics to it's dm* equivalents.

lei137 commented:
gentle ping

RolandF77 commented on llvm/lib/Target/PowerPC/PPCInstrMMA.td:
```diff
@@ -535,25 +535,25 @@ let Predicates = [MMA, IsNotISAFuture] in {
-  // and so register allocation would have to know to match 4 vsr registers
-  // with one wacc register.
-  // On top of that Future CPU has a more convenient way to move between vsrs
-  // and wacc registers using xxextfdmr512 and xxinstdmr512.
```
I don't think we should add patterns for DMXXMTACC/DMXXMFACC for ISA Future, and I don't think we should change the comment. Lowering will replace these intrinsic calls.

lei137 commented on llvm/lib/Target/PowerPC/PPCRegisterInfo.td:
```diff
@@ -20,6 +20,8 @@ def sub_64 : SubRegIndex<64>;
 def sub_vsx0 : SubRegIndex<128>;
 def sub_vsx1 : SubRegIndex<128, 128>;
+def sub_vsx2 : ComposedSubRegIndex<sub_vsx1, sub_vsx0>;
+def sub_vsx3 : ComposedSubRegIndex<sub_vsx2, sub_vsx0>;
```
These are not part of this PR and I will remove them.

lei137 committed:
Address Roland's concern that XXMTACC/XXMFACC are awkward and probably
not handled properly. They have been superseded by new instructions that
can do the same thing, so we don't actually want to create them.

RolandF77 reviewed (approved):
LGTM
